static/specification.html

```diff
@@ -138,605 +336,555 @@ <h2>
         <li>Developers have a way to associate payments with their users, in
         order to provide a range of experiences.
         </li>
-        <li>Pays continuously as the user consumes content.
+        <li>Users retain control of if, when, and how payments are made to the
+        site. For example, micropayments of a predetermined monetary value
+        could be "streamed" to the site over time. Alternatively, the user
+        could make one or many discreet "one-off" payment(s) of any arbitrary
+        monetary amount, even while micropayments are simultaneously being
+        streamed.
         </li>
       </ul>
     </section>
     <section>
       <h2>
-        Extensions to the `Document` interface
+        Link type "monetization"
       </h2>
-      <pre class="idl">
-        partial interface Document {
-          readonly attribute Monetization monetization;
-        };
-      </pre>
+      <aside class="issue">
+        <p>
+          This is a "monkey patch" for the [[HTML]] specification: If accepted,
+          this will eventually be moved to HTML itself.
+        </p>
+      </aside>
+      <aside class="note" title="Multiple monetization links">
+        <p>
+          Although a single HTML document can have multiple [^link^] elements
+          with the "monetization" link type, the user agent might restrict the
+          number of concurrent [=payment sessions=] it can simultaneously
+          support. In such a case, an "error" event will be fired at the
+          [^link^] element where a connection can't be established.
+        </p>
+      </aside>
+      <p>
+        The monetization keyword indicates a [=payment end-point=] used to
+        monetize the document.
+      </p>
+      <p>
+        The <code data-x="rel-monetization">monetization</code> keyword may be
+        used with <code>link</code> elements. This keyword creates an
+        <span data-x="external resource link">external resource link</span>.
+      </p>
+      <p>
+        The <code data-x="rel-monetization">monetization</code> keyword
+        indicates that some aspect of the document is monetized.
+      </p>
+      <div>
+        <p>
+          The default type for resources given by the <code data-x=
+          "rel-monetization">monetization</code> keyword is
+          <code>application/json</code>.
+        </p>
+        <p>
+          The appropriate times to <span data-x=
+          "fetch and process the linked resource">fetch and process</span> this
+          type of link is:
+        </p>
+        <ul>
+          <li>
+            <p>
+              When the <span>external resource link</span> is created on a
+              <code>link</code> element that is already <span>browsing-context
+              connected</span>.
+            </p>
+          </li><!-- e.g. rel="" changed, href="" set... -->
```

Comment:
### States

One of the WM2 drafts had states as so (definitions my own)

#### * idle
The link has not been processed

Payments can not come

#### * interactive
Web Monetization is enabled, provider configured, and the link has been 
processed.

Payments /may/ come

(This has been referred to as the payment pointer being validated.
It's unclear if that meant making a request, or simply checking if
it was well formed, though I believe it was the latter)

From [this comment](https://github.com/WICG/webmonetization/issues/45#issuecomment-983013053)

    We've changed the flow for the state of "monetized" in the current
    iteration of the spec. The "interactive" state triggers as soon as
    we validate the payment pointer, hence providing a middle solution
    here. 

    We're showing content as soon as the first payment can be made
    (which is similar to "you have a web monetization provider"),
    instead of showing it only after the first payment has been made.

#### * error

A non recoverable error has occurred which can only be remedied by
reconfiguring the link tag (e.g. new href)

Payments can not come

#### Valid transitions:
* idle -> error
* idle -> interactive
* interactive -> error

#### Link events with implied state

Upon moving to multiple link tags, the state is implied by events:
Link state starts in idle

#### * 'load' event
transitions state to interactive

#### * 'error' event
transitions state to error 

Per tag configuration there is only ever one load and/or error event, so it
fits with the previous model's valid transitions.

In the move from explicit state to implied, it seems interactive has changed
definition from simply validating the payment pointer (check it's well
formed) to actually requesting it.

Given 'load' does not actually map to how the "loading" of a payment pointer is
actually done ([1]) then why not emit 'load' as soon as the payment pointer has
been deemed well formed. If the SPSP (or OP equivalent) fails, then transition
to the error state by emitting an error. 

Most of the time a configured payment pointer will be valid, and most of the
time the payment pointer will be accessible via network. Why delay the
interactive state for edge cases?

[1] Where it may be loaded more than once, or actually not at all via the
link network machinery, but rather by some other component, or at a WM provider
backend. 

### Where this model may break down

Automatic payment is rejected due to disabling

####  You don't emit the load event

State is not 'interactive'

'monetization' event may still come from user initiated tip, even
 though this breaks the state model, where it says no payments will
 come when the link is in the idle state

[Example](https://webmonetization-preview.netlify.app/docs/exclusive-content#code) using only load event to unlock exclusive content

#### You do emit the load event, to keep state coherent

In this case automatic payments are disabled

This means that the load event is not as useful for guessing whether
payments will come, and "optimistically" show content.

Reply:
> How does disabled monetization fit into this ? Just load anyway, and emit 'load' event?

No. If monetization is disabled, don't emit an event

> What about requiring connection details more than once per tag configuration ?

Getting the connection details a second time shouldn't emit a new load event. The load event is related to processing the payment pointer.

I realise this is problematic when trying to do client-side accounting using receipts but this is yet another reason why we want to deprecate receipts.

Reply:
> The load event is related to processing the payment pointer.

In the spirit of optimism, can we emit it earlier than having actually made a network request ?

> Given 'load' does not actually map to how the "loading" of a payment pointer is
actually done ([1]) then why not emit 'load' as soon as the payment pointer has
been deemed well formed. If the SPSP (or OP equivalent) fails, then transition
to the error state by emitting an error.

^

Reply:
I think the outstanding question here is what to do in the case of automatic streaming being disabled by the user or provider but then the user sends a payment explicitly.

Does the browser emit a load event and then a monetization event or just a monetization event?
I suggest we get some implementation experience with tips before we try to make a final decision.